backend/app/controllers/users.js

```diff
@@ -408,83 +409,96 @@ module.exports = {
-                'roleID': req.body.roleID, //user
-                'salt': crypto.randomBytes(16).toString('hex'),
-                'password': User.hashPassword(salt, pass),
-                'isActive': false,
-                activationToken,
-                'organizationId': org.id,
-                'isAnonymous': req.body.isAnonymous ? true : false,
-                'notifyLevel': req.body.notifyLevel,
-                'authId': 0,
-            };
-
-            if (userExistOnAuth.statusCode === 200) {
-                newClient.authId = typeof userExistOnAuth.body === 'string' ? JSON.parse(userExistOnAuth.body).id : userExistOnAuth.body.id;
-            }
+            // if the user didn't exist, or exists but is not active, send an invitation
+            if (!isExistUser || !isExistUser.isActive) {
+
+                const activationToken = crypto.randomBytes(32).toString('hex');
+                const salt= crypto.randomBytes(16).toString('hex');
+                const pass = crypto.randomBytes(5).toString('hex');
+
+                // create the user if it doesn't exist
+                let userObject = isExistUser;
+                let newClient;
+                if (!isExistUser) {
+
+                    newClient = {
+                        'firstName': req.body.firstName,
+                        'lastName': req.body.lastName,
+                        'email': req.body.email,
+                        'roleID': req.body.roleID, //user
+                        'salt': crypto.randomBytes(16).toString('hex'),
+                        'password': User.hashPassword(salt, pass),
+                        'isActive': false,
+                        activationToken,
+                        'organizationId': org.id,
+                        'isAnonymous': req.body.isAnonymous ? true : false,
+                        'notifyLevel': req.body.notifyLevel,
+                        'authId': 0,
+                    };
+
+                    if (userExistOnAuth.statusCode === 200) {
+                        newClient.authId = typeof userExistOnAuth.body === 'string' ? JSON.parse(userExistOnAuth.body).id : userExistOnAuth.body.id;
+                    }
 
-            const userObject = yield thunkQuery(User.insert(newClient).returning(User.id));
-            newClient.id = _.first(userObject).id;
+                    userObject = yield thunkQuery(User.insert(newClient).returning(User.id));
+                    userObject = _.first(userObject);
+                    newClient.id = userObject.id;
 
-            if (req.body.projectId) { // insert the user into the projectUserTable
-                yield * common.insertProjectUser(req, _.first(userObject).id, req.body.projectId);
-            }
+                    bologger.log({
+                        req: req,
+                        user: req.user,
+                        action: 'insert',
+                        object: 'users',
+                        entity: _.first(userObject).id,
```

Comment:
This line is causing issues because we already did this on line 449 above. So we just need to do `entity: userObject.id,` here.